Automattic/sensei
# Add new onboarding structure

Fixes #5642
Fixes #5643

### Changes proposed in this Pull Request

* It updates the onboarding structure. In order to avoid deprecating a lot of methods and classes, and continued with the name "setup wizard" for the code. I think we are just calling it onboarding in the project organization, right?
* This PR also introduces the new Welcome step. When the user clicks on "Get started" it creates the Sensei pages behind the scenes (similar to the previous behavior).
* I already added a basic step for the usage tracking (_"Help us improve your Sensei experience"_), so we reuse the implementation we had in the previous Welcome screen.
* It already introduces some animations, but I created a separate issue for the more complex animations we will have during the onboarding work: https://github.com/Automattic/sensei/issues/5693
* Some steps continue with the old behavior, and design. It will be changed in future PRs.
* This PR also removes some code that won't be needed anymore because of the new design/behavior.
* It also fixes some E2E tests for the first steps, since it has changed.

### Testing instructions


* Navigate to `/wp-admin/admin.php?page=sensei_setup_wizard`.
* Check that the new styles were applied to the onboarding.
* Navigate through the steps, and make sure the navigation is working properly.
* In the usage tracking step, test the 2 options, and after submitting, check the "WP Admin > Sensei LMS > Settings" to see if the option "Enable usage tracking" was updated.


### New/Updated Hooks

* `sensei.setupWizard.welcomeTitle` - JavaScript hook to customize welcome title.
* `sensei.setupWizard.welcomeParagraph` - JavaScript hook to customize welcome paragraph.


### Screenshot / Video

https://user-images.githubusercontent.com/876340/190717281-37dd86c9-ed39-45de-91fc-06bb7f9a571a.mov
